### ISSUE TITLE
Ideas to encourage teaming in InfClass

### ISSUE THREAD

**Issue by rattbag** [necropotame/teeworlds-infclass#49]
I think the best sort of update to InfClass would be the sort which makes camping have more challenge, and which encourages teamwork.

**Engineer**

* Engineer walls are solid to zombies, but don't kill, meaning they rely on Soldiers.

* Engineer can wall jump

* Engineer’s rifle is automatic, and can fire faster. 

The Engineer now relies more on the Soldier to defend bases, but can kill zombies on their own with their laser rifle. They have wall jumping to make them more interesting to play 

**Soldier**

* Soldier has a cooldown before replacing the mines, after they detonated their last one.

* Soldier has longer hook length

* Soldier’s grenades do more damage than usual

Soldier now relies on the Ninja and Medic to protect them while they replace their mines. Their hook now gives them better movement, and they can pull zombies into their traps.

**Medic**

* Medic has a triple jump

This makes them slightly more interesting to play 

**New super zombie class - Feral** 

* Grants a buff to any zombie within proximity. This increases the length of their hook, and increases their acceleration.

Hooks are invaluable for killing basers. Acceleration would be useful for dodging attacks and pulling hooked players. This zombie would be very useful as a support class, and therefore a target for the defenders.

**New zombie class - Ironskin**

* Immune to mines, and can go through laser walls. 

* Cannot attack with the hammer.

These zombies can breach defences, but not attack themselves. They are useful for hooking players and throwing them below. Defenders have to shoot them with their guns.

\- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 
*The changes above would work on their own, even if you think the rest of this below wouldn't work*
\- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

**New class - Scout/Scavenger** 

* The Scout has a Revolver that has 6 shots. It does 5 damage per shot, has a low rate of fire, and knocks out the target.

* It's hammer ability is to create a shield that negates all incoming damage for the next 5 seconds.

* The scout is fast in the air and on land, and its hook is longer than usual.

* The scout is the only class that can see pickups around the map. These look like the laser ammo icon. The Scout can run across these to pick them up, after which they will revolve around their tee. 

* The scout has a rifle. Using this will place a generator. Generators power up the abilities of the other classes. They require constant refills of pickups to function, and zombies can destroy them. Scouts use pickups to refill their gun.

**Generators**

Each class powers up their abilities when they touch a generator

* Every class gets more speed and maneuverability.

* Engineers get a longer laser wall, and a more powerful laser gun.

* Soldiers get more explosive range on their grenades and mines.

* Medics get more spread and damage on their shotgun.

* Ninjas get less cooldown between attacks, and more pistol damage.

* Scouts get a Revolver refill.

To recap, the Engineer relies on the Soldier for defence. The Soldier relies on Medics and Ninjas to cover him while he replaces his mines. Everyone relies on the Medic for healing, and everyone relies on the Scout for powering up their abilities. The Ninja relies on the Scout even more than the other classes, because they’re a combat class.

**Comment by TheRealOrangus** [comment]
#### Opinion

> Engineer walls are solid to zombies, but don't kill, meaning they rely on Soldiers.

Easy to hook close and infect.

> Engineer can wall jump

Easy to camp on iron mountains where zombies can't get them.

> Soldier has a cooldown before replacing the mines, after they detonated their last one

Isn't this hard enough to re-place bombs when there are army of zombies behind it?

> ... and they can pull zombies into their traps.

Isn't the point of the game is to survive?

> Medic has a triple jump

To catch them then would be as hard as to catch ninjas. _annoying for players_

> Grants a buff to any zombie within proximity. This increases the length of their hook, and increases their acceleration.

I like the idea.

> New zombie class - Ironskin

By the description I'd say it is too much overpowered.

> The scout has a rifle. Using this will place a generator. Generators power up the abilities of the other classes. They require constant refills of pickups to function, and zombies can destroy them. Scouts use pickups to refill their gun.

Looks like you have played the Alien mod.

#### Conclusion

Interesting, but too much complicated, shouldn't this be a KISS(Keep It Simple Stupid) mod?
Alien mod is really awesome, but it's complicated, how much people play there? Indeed, 99% of time 0 players.

Too much teamwork, can't explain why, but it would be bad for popularity of the server, like there's a mod called ANTS where A LOT of teamwork is required to win, and people almost never play there because if they enter there's no team, and even if there is, other mates are not likely to work in a team.

**Comment by rattbag** [comment]
> Easy to hook close and infect.

As it is currently, engineers would have to stay out of hook range.

> Easy to camp on iron mountains where zombies can't get them.

There would only be a certain amount of times you can wall jump, much like the ninja's attack.

> Isn't this hard enough to re-place bombs when there are army of zombies behind it?

Not really. Soldiers are already pretty overpowered in that regard. Engineers have to click in two different places to place their wall, whereas soldiers can just swing in and easily guard an area. If they are guarding with allies, this would mean they would have to rely on their attacks, but if they're guarding alone, they'd have to retreat a little way.

> Isn't the point of the game is to survive?

Yes, and eliminating zombies is one way to do that.

> To catch them then would be as hard as to catch ninjas. *annoying for players*

Possibly they would be a little OP.

> By the description I'd say it is too much overpowered.

The Ironskin zombie would rely entirely upon whether they can successfully hook the enemy player. They have no attacks of their own, and would be next to useless against anything but an engineer.

> Looks like you have played the Alien mod.

These wouldn't be all that complex. Shoot a laser to create a generator. Run over pickups, and they perhaps revolve around you. Run over the generator to drop them off.

**Comment by TheRealOrangus** [comment]
> As it is currently, engineers would have to stay out of hook range.

What about replacing the wall then? It would require huge retrieve every time.

> Not really. Soldiers are already pretty overpowered in that regard. Engineers have to click in two different places to place their wall, whereas soldiers can just swing in and easily guard an area. If they are guarding with allies, this would mean they would have to rely on their attacks, but if they're guarding alone, they'd have to retreat a little way.

This way is not little, but long, because zombies would hook the soldier and because soldier cannot explode the bomb just after placing it, otherwise he would kill himself.

> Yes, and eliminating zombies is one way to do that.

Even thought there's infinite amount of back zombies(I mean they have infinite health and will always respawn over and over) during the round?

> The Ironskin zombie would rely entirely upon whether they can successfully hook the enemy player. They have no attacks of their own, and would be next to useless against anything but an engineer.

To hook an enemy successfully? This is quite easy to do in my opinioin.
